packages/context/src/signalr/SignalRConnectionProvider.tsx

```diff
@@ -35,142 +33,17 @@ export const SignalRConnectionContext = createContext<SignalRConnectionContextTy
-                    setTimeout(() => {
-                        if (!documentsConnectionStopped) {
-                            startConnection(connection, numOfAttempts + 1);
-                        }
-                    }, 2000);
-                } else {
-                    console.warn(`Cannot start connection to SignalR due to error "${error}"`);
-                }
-            });
-        };
-
-        if (documentsSignalRConnection) {
-            startConnection(documentsSignalRConnection, 0);
-        }
-
-        return () => {
-            documentsSignalRConnection?.stop();
-            documentsConnectionStopped = true;
-        };
-    }, [documentsSignalRConnection]);
-
-    useEffect(() => {
-        let instancesConnectionStopped = false;
-
-        const startConnection = async (connection: HubConnection, numOfAttempts: number) => {
-            await connection.start().catch((error: Error) => {
-                if (numOfAttempts < 4 && !instancesConnectionStopped) {
-                    setTimeout(() => {
-                        if (!instancesConnectionStopped) {
-                            startConnection(connection, numOfAttempts + 1);
-                        }
-                    }, 2000);
-                } else {
-                    console.warn(`Cannot start connection to SignalR due to error "${error}"`);
-                }
-            });
-        };
-
-        if (instancesSignalRConnection) {
-            startConnection(instancesSignalRConnection, 0);
-        }
-
-        return () => {
-            instancesSignalRConnection?.stop();
-            instancesConnectionStopped = true;
-        };
-    }, [instancesSignalRConnection]);
-
     return (
         <SignalRConnectionContext.Provider
             value={{
-                documentChanges: documentsSignalRConnection
-                    ? {
-                          subscribe: (topicName, callback) => documentsSignalRConnection.on(topicName, callback),
-                          unsubscribe: (topicName, callback) =>
-                              callback
-                                  ? documentsSignalRConnection.off(topicName, callback)
-                                  : documentsSignalRConnection.off(topicName),
-                      }
-                    : undefined,
-                instanceChanges: instancesSignalRConnection
-                    ? {
-                          subscribe: (topicName, callback) => instancesSignalRConnection.on(topicName, callback),
-                          unsubscribe: (topicName, callback) =>
-                              callback
-                                  ? instancesSignalRConnection.off(topicName, callback)
-                                  : instancesSignalRConnection.off(topicName),
-                      }
-                    : undefined,
+                documentChanges: {
+                    subscribe: () => {},
```

Comment:
@actong Where is the function logic being moved to?

Reply:
Not following, what do you mean by function logic?

Reply:
where are we defining the subscribe / unsubscribe methods?

Reply:
These are the subscribe/unsubscribe functions.  Or if you mean the type definition, that's defined above by `SignalRConnectionContextType`.

Reply:
oh nvm. I realise now this is just for backward compatibility.

Reply:
I suppose another option would be to have `useSignalRConnection` actually use `useNotification` under the hood, in which case it would still be functional.

Reply:
that's a good idea -- not sure if I would hold this up for that though

Reply:
I went ahead and implemented it since this means we don't have to worry so much about warning users to change their widgets.  Note I changed the verbiage back to deprecated.